tests/integration/test_integration.py

```diff
@@ -106,3 +107,75 @@ async def test_given_correct_config_when_deploy_and_scale_then_status_is_active(
         await ops_test.model.applications[APPLICATION_NAME].scale(1)
 
         await ops_test.model.wait_for_idle(apps=[APPLICATION_NAME], status="active", timeout=1000)
+
+    async def test_given_tls_requirer_is_deployed_and_related_then_certificate_is_created_and_passed_correctly(  # noqa: E501
+        self, ops_test, charm, cleanup
+    ):
+        certificate = self.get_certificate_from_file(filename="tests/certificate.pem")
+        ca_certificate = self.get_certificate_from_file(filename="tests/ca_certificate.pem")
+        ca_chain = self.get_certificate_from_file(filename="tests/ca_chain.pem")
+        certificate_bytes = base64.b64encode(certificate.encode("utf-8"))
+        ca_certificate_bytes = base64.b64encode(ca_certificate.encode("utf-8"))
+        ca_chain_bytes = base64.b64encode(ca_chain.encode("utf-8"))
+        config = {
+            "certificate": certificate_bytes.decode("utf-8"),
+            "ca-chain": ca_chain_bytes.decode("utf-8"),
+            "ca-certificate": ca_certificate_bytes.decode("utf-8"),
+        }
+        await ops_test.model.deploy(
+            entity_url=charm,
+            application_name=APPLICATION_NAME,
+            config=config,
+            series="jammy",
+        )
+
+        await ops_test.model.wait_for_idle(apps=[APPLICATION_NAME], status="active", timeout=1000)
+
+        await ops_test.model.deploy(
+            TLS_REQUIRER_CHARM_NAME,
+            application_name=TLS_REQUIRER_CHARM_NAME,
+            channel="edge",
+        )
+        await ops_test.model.add_relation(
+            relation1=f"{APPLICATION_NAME}:certificates", relation2=f"{TLS_REQUIRER_CHARM_NAME}"
+        )
+        await ops_test.model.wait_for_idle(
+            apps=[TLS_REQUIRER_CHARM_NAME],
+            status="active",
+            timeout=1000,
+        )
+        action_output = await run_get_certificate_action(ops_test)
+        assert action_output["certificate"] == self.get_certificate_from_file(
+            filename="tests/certificate.pem"
+        ).strip("\n")
+        assert action_output["ca-certificate"] == self.get_certificate_from_file(
+            filename="tests/ca_certificate.pem"
+        ).strip("\n")
+        formatted_chain = (
+            action_output["chain"]
+            .replace("[", "")
+            .replace("]", "")
+            .replace("'", "")
+            .replace(", ", "\n")
+            .replace("\\n", "\n")
+        )
+        assert formatted_chain == self.get_certificate_from_file(
+            filename="tests/ca_chain.pem"
+        ).strip("\n")
```

Comment:
No need to reload the certs from file here:
```suggestion
        assert action_output["certificate"] == certificate.strip("\n")
        assert action_output["ca-certificate"] == ca_certificate.strip("\n")
        formatted_chain = (
            action_output["chain"]
            .replace("[", "")
            .replace("]", "")
            .replace("'", "")
            .replace(", ", "\n")
            .replace("\\n", "\n")
        )
        assert formatted_chain == ca_chain.strip("\n")
```